drivers/clock_control/clock_control_mcux_ccm_rev2.c

```diff
@@ -157,19 +166,25 @@ static int mcux_ccm_get_subsys_rate(const struct device *dev,
 #endif
 
 #ifdef CONFIG_I2S_MCUX_SAI
+#if (defined(CONFIG_SOC_MIMX9352) || defined(CONFIG_SOC_MIMX9131) || defined(CONFIG_SOC_MIMX9111))
 	case IMX_CCM_SAI1_CLK:
-		clock_root =  kCLOCK_Root_Sai1;
+		clock_root = kCLOCK_Root_Sai1 + instance;
+		break;
+#else
+	case IMX_CCM_SAI1_CLK:
+		clock_root = kCLOCK_Root_Sai1;
 		break;
 	case IMX_CCM_SAI2_CLK:
-		clock_root =  kCLOCK_Root_Sai2;
+		clock_root = kCLOCK_Root_Sai2;
 		break;
 	case IMX_CCM_SAI3_CLK:
-		clock_root =  kCLOCK_Root_Sai3;
+		clock_root = kCLOCK_Root_Sai3;
 		break;
```

Comment:
Please don't mix code cleanups with new feature. Create a separate patch for cleanups.